Assets/Scripts/Models/Tile.cs

```diff
@@ -124,25 +124,51 @@ public void UnregisterTileTypeChangedCallback(Action<Tile> callback)
         cbTileChanged -= callback;
     }
 
+    private static int[][] offsets = new int[][] { new int[] { 1, 0 }, new int[] { 0, 1 }, new int[] { -1, 0 }, new int[] { 0, -1 } };
+
     public bool UnplaceFurniture()
     {
         // Just uninstalling.  FIXME:  What if we have a multi-tile furniture?
+        List<Tile> affectedTiles = new List<Tile>();
+
 
         if (furniture == null)
             return false;
 
         Furniture f = furniture;
 
+        f.CancelJobs (); // Canceles all jobs by this furniture.
+
         for (int x_off = X; x_off < (X + f.Width); x_off++)
         {
             for (int y_off = Y; y_off < (Y + f.Height); y_off++)
             {
 
                 Tile t = World.current.GetTileAt(x_off, y_off);
+                if (t.furniture.linksToNeighbour)
+                    affectedTiles.Add (t);
```

Comment:
I think you can discard the affectedTiles list  and just call t.cbOnChanged(t.furniture) right here